src/ffi.rs

```diff
@@ -33,9 +33,11 @@ pub unsafe extern "C" fn open(
     user_data: *mut c_void,
     o_cb: extern "C" fn(*mut c_void, FfiResult),
 ) {
-    catch_unwind_cb(user_data, o_cb, || {
+    catch_unwind_cb(user_data, o_cb, || -> Result<()> {
         let uri = (CStr::from_ptr(uri).to_str()?).to_owned();
-        rust_open(uri)
+        let res = rust_open(uri);
```

Comment:
this should be `rust_open(uri)?` and the result for the `catch_unwind_cb` should be just `Ok(())`